README.md

```diff
@@ -1,132 +1,355 @@
+
+
+2. Error in DLP API : "Too many findings in request"
+
+    DLP has a max findings per request [limit](https://cloud.google.com/dlp/limits#content-redaction-limits) of 3000. Run the pipeline again with a smaller batch size.
+
+
+3. DLP API request quota exhausted
+
+    This can happen if the Dataflow pipeline is being run with a small batch size. Rerun the pipeline with a larger batch size.
+
+    If the batch size cannot be increased or if the issue persists after you saturate the batch size:
+
+   * Increase the value of the dlpApiRetryCount parameter
+
+   * The dlp-dataflow-deidentification repo offers a parameter numShardsPerDLPRequestBatching. Reducing this below the default (100) will reduce the number of parallel requests sent to DLP.
+
+   * Check whether there are other pipelines/clients generating DLP API requests.
+
+   * Submit a request to increase the quota.
+
+
 ## Dataflow DAG
 
-For Deid and Inspect:
+For inspection and de-identification:
+![v2_dag_](diagrams/dlp_dag_new.png)
 
-![v2_dag_](diagrams/dlp_dag_new.png)	   	
 
 
+For re-identification
 
-For Reid:
+![v2_dag_](diagrams/dlp_reid_dag.png)
 
-![v2_dag_](diagrams/dlp_reid_dag.png)	   	
 
 
+## Advanced topics
 
-## Trigger Pipeline Using Public Image
-You can use the gcloud command to trigger the pipeline using Dataflow flex template. Below is an example for de-identification transform from a S3 bucket.
+You can use the gcloud command to trigger the pipeline using a Dataflow flex template. The following example runs a de-identification transformation on data from an S3 bucket.
 
 ```
 gcloud beta dataflow flex-template run "dlp-s3-scanner-deid-demo" --project=<project_id> \
 --region=<region> --template-file-gcs-location=gs://dataflow-dlp-solution-sample-data/dynamic_template_dlp_v2.json \
 --parameters=^~^streaming=true~enableStreamingEngine=true~tempLocation=gs://<path>/temp~numWorkers=5~maxNumWorkers=5~runner=DataflowRunner~filePattern=<s3orgcspath>/filename.csv~dataset=<bq_dataset>~autoscalingAlgorithm=THROUGHPUT_BASED~workerMachineType=n1-highmem-8~inspectTemplateName=<inspect_template>~deidentifyTemplateName=<deid_template>~awsRegion=ca-central-1~awsCredentialsProvider=$AWS_CRED~batchSize=100000~DLPMethod=DEID
 
 ```
-## To Do
-- take out first row as header before processing 
 
+To avoid repeatedly running Gradle, you can [create a flex template](https://cloud.google.com/dataflow/docs/guides/templates/using-flex-templates).
+
+
+## Disclaimer
 
+The pipeline may throw errors/exceptions when provided with large amounts of input data. For this, the user may need to 
+tune the [pipeline parameters](#pipeline-parameters) as per the requirements.
```

Comment:
We should probably rename this as 
**Some considerations**
The pipeline behaviour is dependent on factors like length of record, number of findings per record, DLP API quota on the project, other applications/pipelines generating DLP API traffic etc.,
Customers may have to tweak the parameters documented above and should refer to the troubleshooting section when they observe errors. 
Most errors seen in the pipeline are an indication that the parameters need to be adjusted.
There may be error scenarios that the pipeline doesn't handle yet and may need code changes.
Due to the reasons mentioned above, one should not assume that this solution is production ready.
We should review the wording with @imadethis and @catherinerosesantos

Reply:
Added new section. But needs to be reviewd